newIDE/app/src/ObjectEditor/Editors/SpriteEditor/SpritesList.js

```diff
@@ -333,8 +323,8 @@ const SpritesList = ({
     ]
   );
 
-  const onAddSprite = React.useCallback(
-    async (resourceSource: ResourceSource) => {
+  const applyPointsAndMasksToSpriteIfNecessary = React.useCallback(

```

Comment:
I like this kind of function that you can call and set up the things. Even if it's a bit verbose, it's clearer to find in the code where it's used.